packages/uni_ui/lib/cards/restaurant_card.dart

```diff
@@ -71,21 +72,51 @@ class RestaurantCardHeader extends StatelessWidget {
             child: Text(
               name,
               style: Theme.of(context).textTheme.headlineSmall,
-              overflow: TextOverflow.ellipsis,
+              overflow: TextOverflow.clip,
             ),
           ),
           Expanded(
             flex: 1,
-            child: IconButton(
-                onPressed: onFavoriteToggle,
-                icon: isFavorite
-                    ? Icon(PhosphorIconsFill.heart,
-                        color: Theme.of(context).primaryColor)
-                    : Icon(PhosphorIconsRegular.heart,
-                        color: Theme.of(context).primaryColor)),
+            child: CardFavoriteButton(isFavorite, onFavoriteToggle),
           ),
         ],
       ),
     );
   }
 }
+
+class CardFavoriteButton extends StatefulWidget {
+  const CardFavoriteButton(this.isFavorite, this.onFavoriteToggle, {super.key});
+
+  final bool isFavorite;
+  final VoidCallback onFavoriteToggle;
+
+  @override
+  State<StatefulWidget> createState() {
+    return CardFavoriteButtonState();
+  }
+}
+
+class CardFavoriteButtonState extends State<CardFavoriteButton> {
+  bool isFavorite = false;
+
+  @override
+  void initState() {
+    super.initState();
+    isFavorite = widget.isFavorite;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return IconButton(
+        icon: isFavorite
+            ? UniIcon(UniIcons.heartFill)
+            : UniIcon(UniIcons.heartOutline),
```

Comment:
```suggestion
        icon: UniIcon(
          isFavorite ? UniIcons.heartFill : UniIcons.heartOutline,
        ),
```

Something similar here.